source/manual/data-gov-uk-common-tasks.html.md

```diff
@@ -0,0 +1,250 @@
+
+### Elasticsearch
+
+You shouldn’t normally need to re-instantiate an Elasticsearch service. [PaaS][paas] are responsible for keeping services working. However you can always do it by using cloudfoundry to:
+
+1. Unbind the old Elasticsearch service from the apps that use it.
+2. Delete the service.
+3. Recreate the service.
+4. Rebind the service to the apps and restage them.
+5. Populate it with data from a [PaaS backup][paas-backup] or from the [Publish Data][publish] database via the `search:reindex` task in the repo.
+
+Once set-up on [PaaS][paas] and bound to [Publish Data][publish] and [Find Data][find] everything should work. Environment variables should be provided automatically to the apps.
+
+To repopulate the index with dataset metadata:
+
+* From a PaaS backup: ask [PaaS support](/manual/data-gov-uk-incidents#data-loss).
+* From a Legacy dataset dump: use the `search:reindex` task in [Publish Data][publish].
+
+
+### Redis
+
+Redis is currently the only service that isn’t run on [PaaS][paas] because PaaS currently only has experimental support for Redis. We run our own instance on one of our AWS servers.
+
+To rebuild it from scratch, either:
+
+* Create a new AWS instance, install package redis-server (
+* In `/etc/redis/redis.conf`, set a master password and a port to run on (we use 21112), and change the address binding to 0.0.0.0 to allow connections from outside.
+
+Then the Publish Secrets service needs to be updated to include the information needed for the [Publish Data][publish] app to access it, which is the Host IP, port, and password.
+
+To access the existing AWS instances, ask a dev for the `aws-dd.pem` key. They should also be able to give you access to the data.gov.uk AWS account to create new instances.
+
+Two AWS EC2 instance are used as a Redis server, for staging and production respectively.
+
+To install a more recent version of Redis (>=4):
+
+    sudo add-apt-repository ppa:chris-lea/redis-server
+    sudo apt-get update
+    sudo apt-get install redis-server
+
+### Secrets services
+
+Like other [PaaS][paas] services, standard cloudfoundry commands are used to unbind, delete, create and bind user-provided services. In this case it’s essential to back up the data in the services before deleting it, or get the latest version from the [datagovuk_infrastructure][infrastructure] repository. Recreating them from scratch will require that they eventually include all the variables and secrets that they make available to other apps, as well as the PGP keys of the users who are able to modify those variables.
+
+## Import data from Legacy
+
+While Legacy remains and publishers use it for creating and modifying datasets (including harvested), it is considered the single source of truth for datasets. While normally the data is imported every hour, it is sometimes necessary to reimport everything.
+
+### Populate your local Postgres and Elasticsearch
```

Comment:
There's a bin/import script in publish which does this, which is referred to in the README. So we should be able to lose this section.